java-checks/src/main/java/org/sonar/java/checks/DefaultEncodingUsageCheck.java

```diff
@@ -66,6 +74,44 @@
   private static final String COMMONS_IOUTILS = "org.apache.commons.io.IOUtils";
   private static final String COMMONS_FILEUTILS = "org.apache.commons.io.FileUtils";
 
+  private static final List<MethodMatcher> COMMONS_IO = ImmutableList.of(
+    method(COMMONS_IOUTILS, "copy").parameters(JAVA_IO_INPUTSTREAM, JAVA_IO_WRITER),
+    method(COMMONS_IOUTILS, "copy").parameters(JAVA_IO_READER, JAVA_IO_OUTPUTSTREAM),
+    method(COMMONS_IOUTILS, "readLines").parameters(JAVA_IO_INPUTSTREAM),
+    method(COMMONS_IOUTILS, "toByteArray").parameters(JAVA_IO_READER),
+    method(COMMONS_IOUTILS, "toCharArray").parameters(JAVA_IO_INPUTSTREAM),
+    method(COMMONS_IOUTILS, "toInputStream").parameters(TypeCriteria.subtypeOf(JAVA_LANG_CHARSEQUENCE)),
+    method(COMMONS_IOUTILS, "toString").parameters(BYTE_ARRAY),
+    method(COMMONS_IOUTILS, "toString").parameters("java.net.URI"),
+    method(COMMONS_IOUTILS, "toString").parameters("java.net.URL"),
+    method(COMMONS_IOUTILS, "write").parameters("char[]", JAVA_IO_OUTPUTSTREAM),
+    method(COMMONS_IOUTILS, "write").parameters(TypeCriteria.subtypeOf(JAVA_LANG_CHARSEQUENCE), TypeCriteria.is(JAVA_IO_OUTPUTSTREAM)),
+    method(COMMONS_IOUTILS, "writeLines").parameters("java.util.Collection", JAVA_LANG_STRING, JAVA_IO_OUTPUTSTREAM),
+
+    method(COMMONS_FILEUTILS, "readFileToString").parameters(JAVA_IO_FILE),
+    method(COMMONS_FILEUTILS, "readLines").parameters(JAVA_IO_FILE),
+    method(COMMONS_FILEUTILS, "write").parameters(TypeCriteria.is(JAVA_IO_FILE), TypeCriteria.subtypeOf(JAVA_LANG_CHARSEQUENCE)),
```

Comment:
To cover also subtypes of `CharSequence` . I realized that it would make sense to use subtypes for most of the signatures actually, so I modified other type criteria to use subtypeOf.